utils/near-performance-metrics/src/framed_write.rs

```diff
@@ -0,0 +1,369 @@
+use std::cell::RefCell;
+use std::marker::PhantomData;
+use std::ops::DerefMut;
+use std::pin::Pin;
+use std::rc::Rc;
+use std::task::{Context, Poll};
+use std::{io, task};
+
+use actix::{Actor, ActorContext, ActorFuture, AsyncContext, Running, SpawnHandle};
+use bitflags::bitflags;
+use bytes::BytesMut;
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+use tokio_util::codec::Encoder;
+
+// This file was copied from actix-0.11.0-beta.1/src/io.rs
+// The only change was to add EncoderCallBack
+
+bitflags! {
+    struct Flags: u8 {
+        const CLOSING = 0b0000_0001;
+        const CLOSED = 0b0000_0010;
+    }
+}
+
+const LOW_WATERMARK: usize = 4 * 1024;
+const HIGH_WATERMARK: usize = 4 * LOW_WATERMARK;
+
+#[allow(unused_variables)]
+pub trait WriteHandler<E>
+where
+    Self: Actor,
+    Self::Context: ActorContext,
+{
+    /// Called when the writer emits error.
+    ///
+    /// If this method returns `ErrorAction::Continue` writer processing
+    /// continues otherwise stream processing stops.
+    fn error(&mut self, err: E, ctx: &mut Self::Context) -> Running {
+        Running::Stop
+    }
+
+    /// Called when the writer finishes.
+    ///
+    /// By default this method stops actor's `Context`.
+    fn finished(&mut self, ctx: &mut Self::Context) {
+        ctx.stop()
+    }
+}
+
+struct InnerWriter<E: From<io::Error>, K: EncoderCallBack> {
+    flags: Flags,
+    buffer: BytesMut,
+    error: Option<E>,
+    low: usize,
+    high: usize,
+    handle: SpawnHandle,
+    task: Option<task::Waker>,
+    callback: K,
+}
+
+pub struct EmptyCallBack {}
```

Comment:
why do we need this

Reply:
We don't strictly need `EmptyCallBack `. However, we could make a PR to `actix_util`, with new optional argument to `FrameWrite`, which is the struct used to reporting when buffer gets drained. That way we could completely remove this file, and keep using one from `actix_util`.

I'm also fine with removing `EmptyCallBack` if you prefer.